Test otel metrics with fault tolerance

Yas-Franco opened the pull request:
Test otel metrics with fault tolerance

benjamin-confino committed:
create project for metrics telemetry integration

benjamin-confino committed:
Move from Metric to Telemetry API

benjamin-confino committed:
create project for telemetry 2.0 metrics

benjamin-confino committed:
Move MetricRecorder to OpenTelemetry

benjamin-confino committed:
Create autofeature for fault tolerence and telemetry metrics integration

benjamin-confino committed:
create project for test

benjamin-confino committed:
create test for fault tolerence with telemetry metrics

benjamin-confino committed:
remove test that doesn't run on FaultTolerence 4.0

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/bnd.bnd:
```diff
@@ -0,0 +1,55 @@
+#*******************************************************************************
+# Copyright (c) 2017, 2020 IBM Corporation and others.
```
Copyright date should be 2024

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/bnd.bnd:
```diff
@@ -0,0 +1,55 @@
+	io.openliberty.jakarta.servlet.6.0;version=latest,\
+	com.ibm.websphere.org.eclipse.microprofile.faulttolerance.1.1;version=latest,\
+	com.ibm.websphere.org.eclipse.microprofile.config.1.3;version=latest,\
+	com.ibm.websphere.org.eclipse.microprofile.metrics.1.0;version=latest,\
```
The indentation here is inconsistent

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/build.gradle:
```diff
@@ -0,0 +1,20 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2021 IBM Corporation and others.
```
Copyright date should be 2024

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/bnd.bnd:
```diff
@@ -0,0 +1,59 @@
+#*******************************************************************************
+# Copyright (c) 2018 IBM Corporation and others.
```
2024

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/bnd.bnd:
```diff
@@ -0,0 +1,59 @@
+
+openTelemetryVersion=1.35.0
+
+Bundle-Name: com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0
```
This bundle should probably be called: 

io.openliberty.microprofile.faulttolerance.4.0.telemetry.metrics.2.0

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/src/com/ibm/ws/microprofile/faulttolerance/telemetry/metrics_20/MetricRecorderImpl.java:
```diff
@@ -0,0 +1,37 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
```
2024

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/bnd.bnd:
```diff
@@ -0,0 +1,59 @@
+        com.ibm.ws.microprofile.faulttolerance;version=latest, \
+        com.ibm.ws.microprofile.faulttolerance.spi;version=latest, \
+        com.ibm.websphere.org.eclipse.microprofile.metrics.2.0;version=latest, \
+        com.ibm.websphere.org.eclipse.microprofile.config.1.3;version=latest,\
```
Indentations are inconsistent

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/src/com/ibm/ws/microprofile/faulttolerance/telemetry/metrics_20/MetricRecorderImpl.java:
```diff
@@ -0,0 +1,37 @@
+import com.ibm.ws.microprofile.faulttolerance.spi.TimeoutPolicy;
+
+/**
+ * Records metrics using MP Metrics API 2.0
```
MpTelemetry 2.0 metrics (Not mp metrics)

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/src/com/ibm/ws/microprofile/faulttolerance/telemetry/metrics_20/MetricRecorderProviderImpl.java:
```diff
@@ -0,0 +1,125 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
```
2024

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/publish/servers/CDIFaultToleranceMetricsRemoval/server.xml:
```diff
@@ -0,0 +1,29 @@
+		<feature>mpTelemetry-2.0</feature>
+	</featureManager>
+	
+	<mpMetrics authentication="false"/>
```
Is this still needed?

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/publish/servers/CDIFaultToleranceMetricsRemoval/server.xml:
```diff
@@ -0,0 +1,29 @@
+<!--
+    Copyright (c) 2017, 2020 IBM Corporation and others.
```
2024

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/publish/servers/CDIFaultToleranceCircuitBreakerMetricsRemoval/server.xml:
```diff
@@ -0,0 +1,34 @@
+	</application>
+
+	
+	<mpMetrics authentication="false"/>
```
Is this still needed?

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/fat/src/com/ibm/websphere/microprofile/faulttolerance/metrics/fat/suite/FATSuite.java:
```diff
@@ -0,0 +1,29 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2021 IBM Corporation and others.
```
2024

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/fat/src/com/ibm/websphere/microprofile/faulttolerance/metrics/fat/tests/CDIFallbackTest.java:
```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2021 IBM Corporation and others.
```
2024

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.telemetry/bnd.bnd:
```diff
@@ -0,0 +1,52 @@
+#*******************************************************************************
+# Copyright (c) 2018 IBM Corporation and others.
```
2024

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.telemetry/bnd.bnd:
```diff
@@ -0,0 +1,52 @@
+
+openTelemetryVersion=1.35.0
+
+Bundle-Name: com.ibm.ws.microprofile.faulttolerance.telemetry
```
Should probably be 

io.openliberty.microprofile.faulttolerance.telemetry

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.telemetry/src/com/ibm/ws/microprofile/faulttolerance/telemetry/metrics/integration/package-info.java:
```diff
@@ -0,0 +1,20 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
```
2024

Yas-Franco commented on dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/test-applications/faulttolerancemetrics.jar/src/com/ibm/websphere/microprofile/faulttolerance/metrics/utils/CustomException.java:
```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ * Copyright (c) 2017 IBM Corporation and others.
```
2024

benjamin-confino replied:
I don't think we need to change this one, its copying a file unchanged

benjamin-confino committed:
copyrights

benjamin-confino committed:
Rename com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0 to io.openliberty.microprofile.faulttolerance.4.0.telemetry.metrics.2.0
Rename com.ibm.ws.microprofile.faulttolerance.telemetry to io.openliberty.microprofile.faulttolerance.telemetry
Rename com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics to io.openliberty.microprofile.faulttolerance.cdi_fat_telemetry_metrics

benjamin-confino committed:
set to noship

benjamin-confino commented:
#build
